tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type.bal

```diff
@@ -263,6 +263,36 @@ function testRuntimeTypeNameOfIntersectionType() {
 
 type Error error<record { string message; }>;
 
+type ReadonlyTypeDef1 readonly;
+type ReadonlyTypeDef2 readonly;
+type ReadonlyTypeDef3 readonly;
+
+type ReadonlyObjectType1 object {
+    int atb1;
+    function call() returns int;
+};
+
+type IntersectionType1 ReadonlyTypeDef3 & record {|
+    int a;
+|};
+
+type FooType record {|
+    int a;
+|};
+
+type IntersectionType2 ReadonlyTypeDef3 & FooType;
+type IntersectionType3 FooType & ReadonlyTypeDef3;
+type IntersectionType4 ReadonlyTypeDef3 & ReadonlyObjectType1;
+type IntersectionType5 ReadonlyObjectType1 & ReadonlyTypeDef3;
+
+function testReadonlyIntersection() {
+    assertTrue(IntersectionType1 is readonly);
+    assertTrue(IntersectionType2 is readonly);
+    assertTrue(IntersectionType3 is ReadonlyTypeDef1);
+    assertTrue(IntersectionType4 is ReadonlyTypeDef2);
+    assertTrue(IntersectionType5 is ReadonlyTypeDef3);
```

Comment:
Is this really what we need to test?

This checks if the typedesc is readonly, which is always true.

Reply:
updated in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/e7e3a1a067f08b8981bed89e9012b21d2fc3c9e0).